### PR TITLE
[WIP] fix: block base64 response

### PR DESCRIPTION
**Pull request by Senna46**
Responses of block_hash & proposer_address from Tendermint RPC are base64 encorded.
We wanna convert to bech32, but have not gotten the right values.

スクリーンショット 2022-04-26 130757

### CODE REVIEW

File: projects/explorer/src/app/pages/blocks/block/block.component.ts [hunk 1]
```diff
@@ -28,6 +30,25 @@ export class BlockComponent implements OnInit {
...
+        return Buffer.from(base64Decoded).toString('hex');
+      }),
+    );
+    const proposer$ = this.block$.pipe(
+      map((block) => {
+        console.log(block.block?.header?.proposer_address!);
+        const base64Decoded = Uint8Array.from(
+          Buffer.from(block.block?.header?.proposer_address!, 'base64'),
+        );
+        const publicKey = new proto.cosmos.crypto.ed25519.PubKey({
```
Comment (KimuraYu45z) [inline projects/explorer/src/app/pages/blocks/block/block.component.ts:45]: How about

```suggestion
const proposer = new cosmosclient.ValAddress(base64Decoded);
```

Comment (Senna46) [reply projects/explorer/src/app/pages/blocks/block/block.component.ts:45]: I left it neglected. I will check it out.

**Comment by Senna46** [comment]
@KimuraYu45z 
I converted it to a ValidatorAddress and displayed it, but it doesn't seem to match the one in the Validator.
It looks like this field needs to match Validator's. What do you think?
スクリーンショット 2022-06-20 133227
before https://ununifi.io/explorer/

**Comment by KimuraYu45z** [comment]
Yes I think it must match...let me do more research

**Comment by KimuraYu45z** [comment]
How about using consensus address?
https://github.com/cosmos-client/cosmos-client-ts/blob/main/src/types/address/cons-address.ts

**Comment by Senna46** [comment]
I used consensus address, but it is not match.
The type of `block.block?.header?.proposer_address` is not consensus address, I think.
```
          const proposer = new cosmosclient.ConsAddress(base64Decoded);
          const moniker = valList.validators?.find((val) => {
            const pubkey = cosmosclient.codec.protoJSONToInstance(
              cosmosclient.codec.castProtoJSONOfProtoAny(val.consensus_pubkey),
            ) as proto.cosmos.crypto.ed25519.PubKey;
            const consAddr = cosmosclient.ConsAddress.fromPublicKey(pubkey);
            console.log(consAddr, proposer);
            return consAddr == proposer;
          })?.description?.moniker;
```

**Comment by KimuraYu45z** [comment]
> is not consensus address
This idea is wrong.

The instance of valaddress and consaddress is just a twenty bytes binary data.
We can calculate both of them from twenty bytes binary data, and we can convert the valaddress into the corresponding consaddress.

**Comment by Senna46** [comment]
Idea
TxHash bese64=>Decode =>Hex String
Proposaser base64=>Decode=>bach32 encode `valconpub`

**Pull request closed without merging**